_layouts/fossils.html

```diff
@@ -57,14 +57,14 @@ <h1 class="post-title"><em>{{ page.title | escape }}</em>&dagger;</h1>
     {% assign right = fossil.right | replace: "_", " " %}
 
     {% for rank in site.data.ranks %}
-    {% for rank_hash in site.data.taxonomy[rank] %}
-    {% assign rank_name = rank_hash[1] %}
-    {% if rank_name.species contains left and rank_name.species contains right %}
-    {% assign nn = site[rank] | where: "title", rank_name.taxonomy[rank][0] | first %}
-    <dd>{{ rank | capitalize }}: <a href="{{ nn.url | relative_url }}">{{ nn.title }}</a></dd>
+    {% assign name = rank %}
+    {% if fossil[name] != "NA" %}
+    {% assign nn = fossil[name] %}
+    {% assign url = "taxonomy/" | append: rank | append: "/" | append: nn %}
+    <dd>{{ rank | capitalize }}: <a href="{{ url | relative_url }}">{{ nn }}</a></dd>
```

Comment:
It looks like `fossil[name]` is not defined here. Consider using e.g. ` {{ nn | inspect }} ` or `{{ fossil[name] | inspect }}` to see what jekyll thinks these variables have in them during page builds.